th2_data_services/interfaces/events_tree/events_tree_collection.py

```diff
@@ -137,7 +163,13 @@ def _build_trees(self, nodes: Dict[Optional[str], List[Node]]) -> None:
         nodes.pop(None)
 
         self._roots = [EventsTree(root) for root in roots]
+
         self._detached_nodes = nodes
+        if self._detached_nodes:
+            w = "You have created a ETC with detached events."
+            self._logger.warning(w)
+            warnings.warn(w)
+            self._warnings.append(w)
```

Comment:
It should be in the __init__ by two reasons:
1. Functions shouldn't have side effects.
2. This warning should appear after object init (not during init) because the next step in the init method after build_tress  -- `self.recover_unknown_events` -- it should resolve detached events.

Reply:
Will fixed in next commit.

Reply:
Fixed in https://github.com/th2-net/th2-data-services/commit/0460ed688d47f24a298ed465e68c81b9cfbf7118